diff -ruNa org/partition.cpp fix/partition.cpp
--- org/partition.cpp	2019-07-04 07:39:02.000000000 +0900
+++ fix/partition.cpp	2019-07-19 17:04:32.794173000 +0900
@@ -258,7 +258,7 @@
 	Format_Block_Size = 0;
 	Ignore_Blkid = false;
 	Retain_Layout_Version = false;
-	Crypto_Key_Location = "";
+	Crypto_Key_Location = "footer";
 	MTP_Storage_ID = 0;
 	Can_Flash_Img = false;
 	Mount_Read_Only = false;
diff -ruNa org/partitionmanager.cpp fix/partitionmanager.cpp
--- org/partitionmanager.cpp	2019-07-04 07:39:02.000000000 +0900
+++ fix/partitionmanager.cpp	2019-07-19 17:04:47.526925000 +0900
@@ -466,7 +466,7 @@
 		printf("   Alternate_Block_Device: %s\n", Part->Alternate_Block_Device.c_str());
 	if (!Part->Decrypted_Block_Device.empty())
 		printf("   Decrypted_Block_Device: %s\n", Part->Decrypted_Block_Device.c_str());
-	if (!Part->Crypto_Key_Location.empty())
+	if (!Part->Crypto_Key_Location.empty() && Part->Crypto_Key_Location != "footer")
 		printf("   Crypto_Key_Location: %s\n", Part->Crypto_Key_Location.c_str());
 	if (Part->Length != 0)
 		printf("   Length: %i\n", Part->Length);
